supabase/migrations/20230731105035_add_message_count_function.sql

```diff
@@ -0,0 +1,23 @@
+-- Counts the number of user messages per day, excluding messages that are suggestions in the presets 
+-- (Only excludes the first instance of the preset message, so it is counted if the user clicks it)
+
+CREATE OR REPLACE FUNCTION public.count_user_messages(organisation_id INT, presets TEXT[])
+RETURNS TABLE(date date, "count" bigint) LANGUAGE 'plpgsql'
+AS $BODY$
+BEGIN
+  RETURN QUERY 
+    WITH msgs AS (
+        SELECT DATE("created_at") as date_created, 
+            content, 
+            ROW_NUMBER() OVER (PARTITION BY content ORDER BY "created_at") AS rank_in_group
+        FROM chat_messages
+        WHERE role = 'user' AND org_id = count_user_messages.organisation_id
+    )
+    SELECT date_created as date, COUNT(*) as "count"
+    FROM msgs
+    WHERE NOT (content IN (SELECT UNNEST(presets)) AND rank_in_group = 1)
+    GROUP BY date_created
+    ORDER BY date_created;
+END;
+$BODY$;
```

Comment:
This is amazing!! :love_you_gesture:

Open questions:
1. Do we also want to filter by `conversation_index = 0` (so only including the first user message in a chat history)?
2. Or is it not-too-hard to do what we discussed previously: where we check the `assistant` messages that follow each `user` message to see if there are any commands issued (/endpoints called) and only count it as a query when an endpoint is called?

Or is 1 a bad idea and 2 too difficult?

Reply:
1 this would only increment usage if you refresh the page right? Asking for multiple requests in the same convo wouldn't increment usage?
2 would probably be ok but not sure it's worth it rn. Can add a ticket

Reply:
If 2 isn't too hard, please can we do it that way? :sweat_smile: 

It's important that we bill our users fairly, rather than billing them equally for saying "hi" and for a question which requires 5 GPT calls and 20 API calls :sweat_smile: